File: DistC++/Include/Template/BasicList.cc
/*
 * Copyright (C) 1995, 1996, 1997
 *
 * Department of Computing Science,
 * The University,
 * Newcastle upon Tyne,
 * UK.
 *
 * $Id: BasicList.cc,v 1.1 1997/09/25 15:27:53 nmcl Exp $
 */

#ifndef BASICLIST_CC_
#define BASICLIST_CC_

template <class T>
BasicList<T>::BasicList ()
{
}

template <class T>
BasicList<T>::~BasicList ()
{
}

template <class T>
T* BasicList<T>::remove ( const Uid& findMe )
{
    SimpleListI<T> iter(*this);
    T *th = 0;
    
    while (th = iter())
    {
	if (th && th->get_uid() == findMe)
	{
	    SimpleList<T>::remove(th);
	    
	    return th;
	}
    }
    
    return 0;
}

template <class T>
T* BasicList<T>::lookFor ( const Uid& findMe )
{
    SimpleListI<T> iter(*this);
    T *th = 0;
    
    while (th = iter())
    {
	if (th && th->get_uid() == findMe)
	{
	    return th;
	}
    }
    
    return 0;
}

template <class T>
Boolean BasicList<T>::lookFor ( T *findMe )
{
    return SimpleList<T>::lookFor(findMe);
}

template <class T>
Boolean BasicList<T>::remove ( T *findMe )
{
    return SimpleList<T>::remove(findMe);
}

template <class T>
BasicListI<T>::BasicListI ( const BasicList<T> &list )
			  : SimpleListI<T>(list)
{
}

#endif

File: ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/template/BasicList.java
/*
 * Copyright (C) 1998, 1999, 2000,
 *
 * Arjuna Solutions Limited,
 * Newcastle upon Tyne,
 * Tyne and Wear,
 * UK.  
 *
 * $Id: BasicList.java,v 1.1 2003/06/19 10:50:38 nmcl Exp $
 */

package com.arjuna.ats.internal.arjuna.template;

import com.arjuna.ats.arjuna.common.Uid;

/**
 * A BasicList extends the concept of a SimpleList by
 * searching and storing entries based upon their Uids.
 *
 * @author Mark Little (mark@arjuna.com)
 * @version $Id: BasicList.java,v 1.1 2003/06/19 10:50:38 nmcl Exp $
 * @since JTS 1.0.
 */

public class BasicList extends SimpleList
{

    /**
     * Create a new BasicList.
     */

public BasicList ()
    {
    }

    /**
     * Remove the entry (and return it) which has the specified
     * Uid.
     */

public synchronized ListElement remove (Uid findMe)
    {
	SimpleListIterator iter = new SimpleListIterator(this);
	ListElement th = null;
    
	while ((th = iter.iterate()) != null)
	{
	    if (th.get_uid().equals(findMe))
	    {
		super.remove(th);
	    
		return th;
	    }
	}
    
	return null;
    }

    /**
     * Return the entry which has the specified Uid.
     */

public synchronized ListElement lookFor (Uid findMe)
    {
	SimpleListIterator iter = new SimpleListIterator(this);
	ListElement th = null;
    
	while ((th = iter.iterate()) != null)
	{
	    if (th.get_uid().equals(findMe))
	    {
		return th;
	    }
	}
    
	return null;
    }

}

